src/main/java/io/kestra/plugin/hightouch/AbstractHightouchConnection.java

```diff
@@ -0,0 +1,79 @@
+package io.kestra.plugin.hightouch;
+
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.Task;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpResponse;
+import java.net.http.HttpRequest;
+import java.net.MalformedURLException;
+import java.net.ConnectException;
+import java.nio.charset.StandardCharsets;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+import javax.validation.constraints.NotNull;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public abstract class AbstractHightouchConnection extends Task {
+
+    @Schema(
+        title = "API Bearer token"
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
+    String token;
+
+    protected <REQ, RES> io.micronaut.http.HttpResponse <RES> request(String method, String path, String body, Class<RES> responseType) throws IOException, InterruptedException {
+
+        HttpClient httpClient = HttpClient.newBuilder().build();
+        String baseUrl = "https://api.hightouch.com";
+        ObjectMapper objectMapper = new ObjectMapper() // Jackson ObjectMapper for JSON parsing
+                .registerModule(new JavaTimeModule());
```

Comment:
You should instantiate an object mapper as a field constant to avoid creating a new mapper on each task execution.
You can use `JacksonMapper.ofJson()` to use the same object mapper that is used inside Kestra itself so the serialization will be handled the same.